contracts/JBV3Token.sol

```diff
@@ -0,0 +1,355 @@
+    revert("not available");
+  }
+
+  /** 
+    @notice
+    Migrate v1 & v2 tokens to v3.
+  */
+  function migrate() external {
+    uint256 _fundsToMigrateFromV1;
+    uint256 _fundsToMigrateFromV2;
+    if (address(v1TicketBooth) != address(0)) {
+      // Get a reference to the v1 projects ERC20 tokens.
+      ITickets _v1Token = v1TicketBooth.ticketsOf(projectId);
+      _fundsToMigrateFromV1 = _migrateV1Tokens(_v1Token);
+    }
+
+    if (address(v2TokenStore) != address(0)) {
+      // Get a reference to the v2 project's ERC20 tokens.
+      IJBToken _v2Token = v2TokenStore.tokenOf(projectId);
+      _fundsToMigrateFromV2 = _migrateV2Tokens(_v2Token);
+    }
+
+    uint256 _tokensToMint = _fundsToMigrateFromV1 + _fundsToMigrateFromV2;
+    // mint tokens directly
+    _mint(msg.sender, _tokensToMint);
+  }
+
+  /** 
+    @notice
+    Migrate v1 tokens to v3.
+    @param _v1Token The v1 token instance.
+
+    @return amount of v2 tokens to be migrated
+  */
+  function _migrateV1Tokens(ITickets _v1Token) internal returns(uint256) {     
+    // Get a reference to the migrator's unclaimed balance.
+    uint256 _tokensToMintFromUnclaimedBalance = v1TicketBooth.stakedBalanceOf(msg.sender, projectId);
+
+    // Get a reference to the migrator's ERC20 balance.
+    uint256 _tokensToMintFromERC20s = _v1Token == ITickets(address(0)) ? 0 : _v1Token.balanceOf(msg.sender);
+    
+    // total v3 tokens to mint
+    uint256 v3TokensToMint = _tokensToMintFromERC20s + _tokensToMintFromUnclaimedBalance;
+
+    if (v3TokensToMint == 0)
+      revert INSUFFICIENT_FUNDS();
+
+    // Transfer v1 ERC20 tokens to this contract from the msg sender if needed.
+    if (_tokensToMintFromERC20s != 0)
+      IERC20(_v1Token).transferFrom(msg.sender, address(this), _tokensToMintFromERC20s);
+
+    // Transfer v1 unclaimed tokens to this contract from the msg sender if needed.
+    if (_tokensToMintFromUnclaimedBalance != 0)
+      v1TicketBooth.transfer(
+        msg.sender,
+        projectId,
+        _tokensToMintFromUnclaimedBalance,
+        address(this)
+      );
+    
+    return v3TokensToMint;
+  }
+
+  /** 
+    @notice
+    Migrate v2 tokens to v3.
+    @param _v2Token The v2 token instance.
+
+    @return amount of v2 tokens to be migrated
+  */
+  function _migrateV2Tokens(IJBToken _v2Token) internal returns(uint256) {
+    // Get a reference to the migrator's unclaimed balance.
+    uint256 _tokensToMintFromUnclaimedBalance = v2TokenStore.unclaimedBalanceOf(msg.sender, projectId);
+
+    // Get a reference to the migrator's ERC20 balance.
+    uint256 _tokensToMintFromERC20s = _v2Token == IJBToken(address(0)) ? 0 : _v2Token.balanceOf(msg.sender, projectId);
+    
+    // total v3 tokens to mint
+    uint256 v3TokensToMint = _tokensToMintFromERC20s + _tokensToMintFromUnclaimedBalance;
+
+    if (v3TokensToMint == 0)
+      revert INSUFFICIENT_FUNDS();
```

Comment:
Same, but more of a weird case here

Reply:
I think returning 0 should take care of it